metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,413 @@
+        self.cassandra_session = session
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = CassandraSourceConfig.parse_obj(config_dict)
+        return cls(ctx, config)
+
+    def get_workunits_internal(
+            self,
+    ) -> Iterable[MetadataWorkUnit]:
+
+        # get all keyspaces and iterate through them
+        keyspaces = self.cassandra_session.execute(GET_KEYSPACES_QUERY)
+        keyspaces = sorted(keyspaces, key=lambda k: getattr(k, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["keyspace_name"]))
+
+        for keyspace in keyspaces:
+            keyspace_name = getattr(keyspace, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["keyspace_name"])
+            
+            # skip system keyspaces
+            if keyspace_name in SYSTEM_KEYSPACE_LIST:
+                continue
+            # skip keyspaces excluded by the config
+            if self.config.excludeKeyspaces and keyspace_name in self.config.excludeKeyspaces:
+                self.report.report_dropped(keyspace_name)
+                continue
+
+            # get all tables for this keyspace and emit their metadata
+            try:
+                yield from self._extract_tables_from_keyspace(keyspace_name)
+            except Exception as e:
+                logger.warning(f"Failed to extract table metadata for keyspace {keyspace_name}", exc_info=True)
+                self.report.report_warning(
+                    "keyspace", f"Exception while extracting keyspace tables {keyspace_name}: {e}"
+                )
+            
+    # get all tables for a given keyspace, iterate over them to extract column metadata
+    def _extract_tables_from_keyspace(self, keyspace_name: str) -> Iterable[MetadataWorkUnit]:
+        tables = self.cassandra_session.execute(GET_TABLES_QUERY, [keyspace_name])
+        tables = sorted(tables, key=lambda t: getattr(t, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["table_name"])) # sorted so output is consistent
+        for table in tables:
+            # define the dataset urn for this table to be used downstream
+            table_name = getattr(table, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["table_name"])
+            dataset_name = f"{keyspace_name}.{table_name}"
+            dataset_urn = make_dataset_urn_with_platform_instance(
+                platform=PLATFORM_NAME_IN_DATAHUB,
+                name=dataset_name,
+                env=self.config.env,
+                platform_instance=self.config.platform_instance,
+            )
+
+            # 1. Extract columns from table, then construct and emit the schemaMetadata aspect.
+            try:
+                yield from self._extract_columns_from_table(keyspace_name, table_name, dataset_urn)
+            except Exception as e:
+                logger.warning(f"Failed to extract columns from table {table_name}", exc_info=True)
+                self.report.report_warning(
+                    "table", f"Exception while extracting table columns {table_name}: {e}"
+                )
```

Comment:
if this fails should I bother reporting the status aspect below or just return? i figured if we know about the table we should at least report that.